### DISCUSSION TITLE
Update-2022-10-05

### DISCUSSION THREAD

**Discussion by kaveh808** [kaveh808/kons-9#118] [Announcements]
Greetings, everyone.

I have pushed a new "stand-alone" version of the `text-engine` branch. All the lib code for the fonts has been included in the project explicitly. This is temporary to facilitate downloading and running the branch.

All demos run correctly.

The prototype GUI is very rough, but I'd appreciate if people could give it a go and report their impressions.

I hope to make some changes to the scene data structure (replace the shapes and motions lists with groups to handle hierarchies and the deletion of nodes better) and then make a `developer-preview-1` branch, which will be our first "public" sneak peek.

**Comment by aykaramba** [comment]
// EDIT: Can someone comment on whether or not it is appropriate to post a reply in an UPDATE thread?  I would be happy to delete / move it as per rules.  Thanks.


Hi!

So far, I can report the UI menu boxes render correctly but the text does not either line up properly or render within the boxes.  Screenshot here: 

![screenshot](https://user-images.githubusercontent.com/16431651/194181411-88334ee9-92af-452f-a7ad-75a7b478f355.png)

System info: Linux 5.10.0-18-amd64 #1 SMP Debian 5.10.140-1 (2022-09-02) x86_64 GNU/Linux (Debian 11).  

Feedback; I am a fanboy of Lisp Machines, Mirai and Softimage so I cannot provide any impartial feedback, I love everything that I see.  I have really enjoyed looking through the example files and fiddling with some rudimentary code.  This is pretty much a dream and any rough edges are meaningless to me.  

Having said that, if you want noob level + 3d enthusiast testing, let me know what you want me to test and I would be happy to give it a go.

**Reply by kaveh808** [reply]
Replying here is fine.

I will look into the text alignment issue. It's probably due to you having a "retina" style monitor, where there is a built-in pixel scaling factor.

I appreciate you testing the system. If you like, go through the demos in demo-kernel.lisp and let me know if anything is confusing or there is something you would like explained in more detail.

**Reply by aykaramba** [reply]
Thanks.  Will do.  It's a 4k tv, so close, the dpi and gamut is way off relative to normal monitors.

**Reply by kaveh808** [reply]
Run this and see what it returns:

```
(glfw:get-monitor-content-scale (glfw:get-primary-monitor))
```

**Reply by mikelevins** [reply]
> Having said that, if you want noob level + 3d enthusiast testing, let me know what you want me to test and I would be happy to give it a go.

Speaking only for myself, I would think that having a noob-level enthusiast posting impressions and reactions is one of the most desirable thinkgs we could wish for. It's exactly those kinds of users that are best at catching problems with usability.

**Reply by aykaramba** [reply]
Apologies for the delayed response.  

When I run this: (glfw:get-monitor-content-scale (glfw:get-primary-monitor))
The result is this: (1.1458334 1.1458334 )

My screen resolution is: 3840x2160

I would be happy to perform any tests requested.  Let me know if that is helpful.  I have downloaded the latest text-engine branch and the results are similar to the screenshot above.

**Reply by aykaramba** [reply]
> Speaking only for myself, I would think that having a noob-level enthusiast posting impressions and reactions is one of the most desirable thinkgs we could wish for. It's exactly those kinds of users that are best at catching problems with usability.

I appreciate that.  Following this project has been tremendous fun so far and the code Kaveh writes is so easy to understand it makes it a lot of fun to play with the system.  

I am looking forward to working through this using Kons-9: https://www.youtube.com/user/codingmath/videos
Steve Losh already did it on Sketch, I think it would be neat to do the same in Kons-9: https://github.com/sjl/coding-math

**Reply by kaveh808** [reply]
Please open an issue for the text placement and I will look into it. Thanks for testing this.

**Comment by kaveh808** [comment]
The coding math project looks very cool. One thing you could do in kons-9 is generalize the exercises to 3D. Plus they would be great demos for the system, both as code samples and if you want to make videos.

**Comment by lukego** [comment]
Great hacking, @kaveh808!

I opened PR #121 with some small fixes that I needed to make the `text-engine` branch work on my computer. I hope that's useful.

**Reply by kaveh808** [reply]
Cool, thanks. I added a comment.

**Comment by awolven** [comment]
I guess (based on the commit diff for PR #121) that the text-engine branch
is still using freetype2?

**Reply by kaveh808** [reply]
My version of `opengl3-text.lisp` doesn't have it. Not sure if the PR is referring to an older version.
